pypit/core/artraceslits.py

```diff
@@ -2802,51 +2807,97 @@ def trace_gweight(fimage, xinit_in, sigma = 1.0, ycen = None, invvar=None, maskv
          Formal propagated error on recentroided trace. These errors will only make sense if invvar is passed in, since
          otherwise invvar is set to 1.0.   The output will have the same shape as xinit i.e.  an 2-d  array with shape
          (nspec, nTrace) array if multiple traces were input, or a 1-d array with shape (nspec) for the case of a single
-         trace. Locations where the gaussian weighted centroid deviates from the input guess by  > radius, or where it falls
-         off the image, will have this error set to 999 and will their xnew values set to that of the input trace. These
-         should thus be masked in any fit.
+         trace. Locations where the gaussian weighted centroid falls off the image, will have this error set to 999 and
+         will their xnew values set to that of the input trace. These should thus be masked in any fit via a condition like (xerr < 999)
 
      Revision History
      ----------------
      Python port of trace_fweight.pro from IDLUTILS
-     24-Mar-1999  Written by David Schlegel, Princeton.
+     17-Jan-2000  Written by Scott Burles, Chicago
      27-Jun-2018  Ported to python by X. Prochaska and J. Hennawi
     '''
 
-    # Setup
+
+    # Init
     nx = fimage.shape[1]
-    xnew = np.zeros_like(xcen)
-    xerr = maskval*np.ones_like(xnew)
+
+    # Checks on radius
+    if (isinstance(sigma,int) or isinstance(sigma,float)):
+        sigma_out = sigma
+    elif ((np.size(sigma)==np.size(xinit_in)) & (np.shape(sigma) == np.shape(xinit_in))):
+        sigma_out = sigma
+    else:
+        raise ValueError('Gaussian sigma must a be either an integer, a floating point number, or an ndarray '
+                         'with the same shape and size as xinit_in')
+
+
+    # Figure out dimensions of xinit
+    dim = xinit_in.shape
+    ndim = xinit_in.ndim
+    if (ndim == 1):
+        nTrace = 1
+        npix = dim[0]
+    else:
+        nTrace = dim[1]
+        npix = dim[0]
+
+    ncen = xinit_in.size
+
+    xinit = xinit_in.flatten()
+    # Create xnew, xerr
+    xnew = xinit.astype(float)
+    xerr = np.full(ncen,999.)
+
+    if ycen is None:
+        if ndim == 1:
+            ycen = np.arange(npix, dtype='int')
```

Comment:
I still don't understand why one puts '' around int.
I'd avoid it.

Reply:
That was a bug.